### ISSUE TITLE
Generic.Functions.OpeningFunctionBraceBsdAllman removes return type when additional lines are present

### ISSUE THREAD

**Issue by markstory** [squizlabs/PHP_CodeSniffer#3357]
**Describe the bug**
When a function has a return type and an additional new line between the brace and function signature, `phpcbf` will remove the return type.

**Code sample**
```php
<?php

class FormattingBug
{
    public function extraLine(string: $a): void

    {
        // code here.
    }
}
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcbf --standard=PSR12 test.php`

phpcbf will update the problematic formatting, however the resulting code looks like:

```php
<?php

class FormattingBug
{
    public function extraLine(string: $a)
    {
        // code here.
    }
}
```

**Expected behavior** 

Formatting to be updated without removing the return type.

**Comment by jrfnl** [comment]
@markstory Oh wow! Thanks for reporting this. Very curious to realize nobody has reported this before, even though return types have been used for years now.

Issue is reproducable and confirmed.

While when running with PSR12, the issue points to the `Squiz.Functions.MultiLineFunctionDeclaration.BraceSpacing` error code, the actual issue is in the `Generic.Functions.FunctionOpeningBraceBsdAllman` sniff.

PR #3358 should fix this. Testing appreciated.

**Comment by markstory** [comment]
The output results look good to me! Thank you for the speedy fix :heart:
